plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -194,51 +193,53 @@ private CollectedStatistics mergeStatisticsIfNecessary(
                 updateMode = REPLACE;
             }
         }
+        Map<Integer, CompactSketch> collectedNdvSketches = collectedStatistics.ndvSketches();
+        ImmutableMap.Builder<Integer, CompactSketch> ndvSketches = ImmutableMap.builder();
```

Comment:
why is the builder allocated outside of lambda below, but then populated and built inside lambda?